### ISSUE TITLE
Modal Lightbox and JavaScript

### ISSUE THREAD

**Issue by NoisyMonk** [wet-boew/wet-boew#7919]
Hi all,

I'm having trouble with the Modal Lightbox and Javascript. Basically, the modal popup will close if you click outside the popup, much like a regular overlays, instead of staying open. The close button works fine. Any thoughts?

JS:
```
$(document).trigger("open.wb-lbx", [
        [
            {
                src: "#seach-overlay",
                type: "inline"
            }
        ]
    ]);
```

HTML:
```
<section id="seach-overlay" class="mfp-hide modal-dialog modal-content overlay-def">
    <header class="modal-header">
        <h2 class="modal-title">Search Options</h2>
    </header>
    <div class="modal-body">
        ... Search Options ...
        <button class="btn btn-default btn-sm popup-modal-dismiss">Close</button>
    </div>
</section>
```

**Comment by kodecount** [comment]
Yeah the trigger for some reason can't tell that it a `lbx-modal` (I can't find where to even add this option.

For now you can do:
```javascript
$( document ).ready(function() {
	$( document ).trigger( "open.wb-lbx", [ [ { src:'#centred-popup-modal', type:'inline' } ] ], true );
	setTimeout(function() {
		$( ".mfp-wrap" ).unbind( "click vclick" );
	}, 1000 );
});
```

Also if you change `"open.wb-lbx"` to an invalid trigger like `"open.randomClass"` it breaks the whole page.

**Comment by NoisyMonk** [comment]
I'll give that a shot, thanks!

It's weird though, because it does seem to be possible via the [example here with the select box](https://wet-boew.github.io/v4.0-ci/demos/lightbox/lightbox-en.html#examples5).

**Comment by kodecount** [comment]
I think it due to this: `<a class="wb-lbx lbx-modal" title="Example of a modal popup" href="#inline_content_modal">Modal popup</a>`

**Comment by kodecount** [comment]
Oddly it works using:
``javascript
(function( $, wb ) {
"use strict";

var $document = wb.doc;

$document.trigger( "open.wb-lbx", [
	[
		{
			src: "#centred-popup-modal",
			type: "inline"
		}
	],
	true
]);

})( jQuery, wb );
```
From the example.

**Comment by kodecount** [comment]
Maybe the documentation should be updated to have triggers used in a WET wrapper.

**Comment by NoisyMonk** [comment]
Yep, that works for me - thanks!

Although as you mentioned, their still the issue of ESC key. But that seems like a different can of worms, so this issue can probably be marked as closed.

Cheers.

**Comment by RobJohnston** [comment]
I think that by setting "true", it enabled the [modal option](http://dimsemenov.com/plugins/magnific-popup/documentation.html#modal) of the Magnific Popup.  From the docs, 

> When set to true, the popup will have a modal-like behavior: it won’t be possible to dismiss it by usual means (close button, escape key, or clicking in the overlay).
> 
> This is a shortcut to set closeOnContentClick, closeOnBgClick, showCloseBtn, and enableEscapeKey to false.

**Comment by kodecount** [comment]
Yes, but it only works in this wrapper `(function( $, wb ) {"use strict";})( jQuery, wb );' if I was to trigger in a regular function it won't work.

**Comment by RobJohnston** [comment]
Not sure I believe it...  I think "true" was the only thing missing from the OP's JS function, and it **will** disable the escape key too.  I may have to get a clean install of WET, but that's been my experience.

**Comment by kodecount** [comment]
Yes that how it suppose to work, but it doesn't. Go try it your self.

**Comment by kodecount** [comment]
On a personal project I have to use this function to get an light-box modal to work:

```javascript
function triggerLbxModal( selector ) {
  (function( $, wb ) {
 "use strict";

 var $document = wb.doc;

 $document.trigger( "open.wb-lbx", [ [ { src: selector, type: "inline" } ], true ]);
 })( jQuery, wb );
}
```
